packages/js/src/plugins/auctionHouseModule/operations/directSell.ts

```diff
@@ -0,0 +1,307 @@
+      const { lazyPurchase, listing, response } = await (
+        await directSellBuilder(metaplex, operation.input)
+      ).sendAndConfirm(metaplex, operation.input.confirmOptions);
+
+      const purchase = await metaplex
+        .auctionHouse()
+        .loadPurchase({ lazyPurchase })
+        .run();
+
+      return { listing, purchase, response };
+    },
+  };
+
+// -----------------
+// Builder
+// -----------------
+
+/**
+ * @group Transaction Builders
+ * @category Inputs
+ */
+export type DirectSellBuilderParams = Omit<
+  DirectSellInput,
+  'confirmOptions'
+> & {
+  createListingInstructionKey?: string;
+  executeSaleInstructionKey?: string;
+};
+
+/**
+ * @group Transaction Builders
+ * @category Contexts
+ */
+export type DirectSellBuilderContext = Omit<
+  DirectSellOutput,
+  'response' | 'purchase'
+> & { lazyPurchase: LazyPurchase };
+
+/**
+ * Creates a listing on a given asset and executes a sale on the created listing and given bid.
+ *
+ *
+ * ```ts
+ * const transactionBuilder = metaplex
+ *   .auctionHouse()
+ *   .builders()
+ *   .sell({ auctionHouse, bid, seller })
+ * ```
+ *
+ * @group Transaction Builders
+ * @category Constructors
+ */
+export const directSellBuilder = async (
+  metaplex: Metaplex,
+  params: DirectSellBuilderParams
+): Promise<TransactionBuilder<DirectSellBuilderContext>> => {
+  // Data.
+  const {
+    auctionHouse,
+    auctioneerAuthority,
+    bid,
+    seller = metaplex.identity(),
+    authority = auctionHouse.authorityAddress,
+    bookkeeper = metaplex.identity(),
+    createListingInstructionKey,
+    executeSaleInstructionKey,
+  } = params;
+  const { hasAuctioneer } = auctionHouse;
+  const { tokens, price, buyerAddress, isPublic, asset } = bid;
+
+  const tokenAccount = isPublic
+    ? findAssociatedTokenAccountPda(asset.address, toPublicKey(buyerAddress))
+    : (asset as SftWithToken | NftWithToken).token.address;
+  const printReceipt =
+    (params.printReceipt ?? true) && Boolean(bid.receiptAddress);
+
+  if (hasAuctioneer && !auctioneerAuthority) {
+    throw new AuctioneerAuthorityRequiredError();
+  }
+
+  const listingBuilder: TransactionBuilder<CreateListingBuilderContext> =
+    await createListingBuilder(metaplex, {
+      mintAccount: asset.mint.address,
+      price: price,
+      auctionHouse,
+      auctioneerAuthority,
+      seller,
+      authority,
+      tokenAccount,
+      tokens,
+      printReceipt,
+      bookkeeper,
+      instructionKey: createListingInstructionKey,
+    });
+  const { receipt, sellerTradeState } = listingBuilder.getContext();
+
+  let listingAsset: NftWithToken | SftWithToken;
+
+  if (isNftWithToken(asset) || isSftWithToken(asset)) {
+    listingAsset = asset;
+  } else {
+    // Load asset token if the bid was public and there is no token data in the asset model.
+    const asssetTokenAddress = findAssociatedTokenAccountPda(
+      asset.address,
+      toPublicKey(seller)
+    );
```

Comment:
@zaxozhu Can you remind me why `tokenAccount` uses the buyer address whereas `asssetTokenAddress` uses the seller address?

Reply:
Oh, yeah, that looks off. There must be a seller's token account.